plugin/backend_lookup.go

```diff
@@ -403,4 +403,37 @@ func newAddress(s msg.Service, name string, ip net.IP, what uint16) dns.RR {
 	return &dns.AAAA{Hdr: hdr, AAAA: ip}
 }
 
+func checkZoneForRecord(b ServiceBackend, zone string, state *request.Request, opt Options) ([]msg.Service, error) {
```

Comment:
In line [412](https://github.com/m4r10k/coredns/blob/b189084482f759c3b577c698e0a9be3e4a040c36/plugin/backend_lookup.go#L412) I am saving it in case to restore it as needed in [428](https://github.com/m4r10k/coredns/blob/b189084482f759c3b577c698e0a9be3e4a040c36/plugin/backend_lookup.go#L428) so that it is usable back in line [28](https://github.com/m4r10k/coredns/blob/b189084482f759c3b577c698e0a9be3e4a040c36/plugin/backend_lookup.go#L28) because the `state` is never returned and it must be corrected as otherwise dig will complain that the answer does not match the question. Which I saw in some cases for the `NS` query (I can provide more info tomorrow)